.github/workflows/main.yaml

```
@@ -4,38 +4,49 @@ on:
+        uses: actions/setup-go@v2
         with:
           go-version: '1.23' 
 
-      - name: Docker
-        run: make docker-build
+      - name: Extract Git Tag
+        if: startsWith(github.ref, 'refs/tags/')
+        run: echo "GIT_TAG=${GITHUB_REF#refs/tags/}" >> "$GITHUB_ENV"
+
+      - name: Docker Build
+        run: |
```

actionlint (1.7.3)

shellcheck reported issue in this script: SC2086:info:2:83: Double quote to prevent globbing and word splitting

(shellcheck)